git/dump.py

```diff
@@ -19,3 +20,67 @@ def from_string(cls, type_str):
 
     def __str__(self):
         return '{}'.format(self.name)
+
+class TextDump:
```

Comment:
Could you please extract this `TextDump` class into 3 separated classes e.g.  _default_, _header_ and _custom_ ? I think it could be better, more maintainable and of course more understandable.